File: cs/DressDiscover/DressDiscover.Api/src/DressDiscover/Api/Models/Worksheet/WorksheetStateId.cs
namespace DressDiscover.Api.Models.Worksheet
{
    public sealed class WorksheetStateId {
        public static WorksheetStateId Parse(string value) {
            if (value.Length == 0) {
                throw new System.ArgumentException("id is empty", "WorksheetStateId");
            }
            return new WorksheetStateId(value);
        }

        private WorksheetStateId(string value) {
            this.value = value;
        }

        public override string ToString() {
            return this.value;
        }

        private string value;
    }
}

File: ts/worksheet-knockout/src/dressdiscover/api/models/worksheet/worksheet_state_id.ts
export class WorksheetStateId {
    public static parse(id: string): WorksheetStateId {
        if (id.trim().length == 0) {
            throw new RangeError("id is empty");
        }
        return new WorksheetStateId(id);
    }

    public equals(other?: WorksheetStateId): boolean {
        if (typeof(other) === "undefined") {
            return false;
        }
        return this._id == other._id;
    }

    public toString(): string {
        return this._id;
    }

    private constructor(id: string) {
        this._id = id;
    }

    private _id: string;
}
